source/non_matching/coupling.cc

```diff
@@ -140,6 +136,26 @@ namespace NonMatching
       if (immersed_c[i])
         immersed_gtl[i] = j++;
 
+    const unsigned int n_q_points = quad.size();
+    const unsigned int n_active_c =
+      immersed_dh.get_triangulation().n_active_cells();
+    std::vector<Point<spacedim>> all_points(n_active_c * n_q_points);
+    {
+      unsigned int c = 0;
+      for (const auto &cell : immersed_dh.active_cell_iterators())
+        {
+          // Reinitialize the cell and the fe_values
+          fe_v.reinit(cell);
```

Comment:
I believe this is the only place where you use `fe_v` in this function. Move its declaration to line 144, inside the current `{...}` block.

Reply:
Wow, I would have never though about that. Thanks!

Reply:
I'll even tell you why I suggested this. When I looked at the code, I was wondering what the update flags for the FEValues object were because you were only looking at the quadrature point locations. So I had to scroll up to see where that object was declared. In other words, for a reader of a code, it's *convenient* to have the declaration as far down in a function as possible because then the declaration/initialization and the use of a variable are as close as possible together. 

Secondly, it's useful to put a declaration in the inner-most scope possible so that the variable is destroyed as soon as possible -- so that (i) the memory used up is released again, and (ii) if a reader wonders whether the variable is used elsewhere, one doesn't have to scroll down very far.

In summary, it's generally a good idea to limit the lifetime of local variables as much as possible by pushing declarations down in the function and into the innermost scope possible.